### ISSUE TITLE
[git 2.16.1 x64] Installer is inserting misplaced entries in winreg which replaces explorer as default folders opener

### ISSUE THREAD

**Issue by shadyueh** [git-for-windows/git#1458]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

Since the version 2.15.1.2 x64 of git for windows, its Installer is inserting misplaced entries in Windows registers of my computer which replaces Explorer as default folders opener.
I just downloaded the version 2.16.1 to test if the problem was already solved but it still occurrs.

I'm specifically talking about 
`HKEY_CLASSES_ROOT\Directory\shell\git_gui\command`  whose value is
"C:\Program Files\Git\cmd\git-gui.exe" "--working-dir" "%1"
and
`HKEY_CLASSES_ROOT\Directory\shell\git_gui\command` whose value is
"C:\Program Files\Git\git-bash.exe" "--cd=%1"
By a reason that just dark deities know, this causes **explorer replacement as default folders opener and we can't double click to open the folders**.

I **workarounded** this deleting that entries, but I'm reporting for you to fix the installer.

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
```
$ git --version --build-options

git version 2.16.1.windows.1
cpu: x86_64
built from commit: 1a4ee4d5d83bbba63f6d8b71c427f627769f3386
sizeof-long: 4

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [versão 6.1.7601]
(It's the Windows 7 Professional 64 bits)
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Check the Details below

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Not applies to the problem. **It's related to the instalation process.**

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

**I tried open a folder through double click in explorer.**

 - What did you expect to occur after running these commands?

**Explorer should open the folder**

 - What actually happened instead?

**Git GUI is opened**

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Not applies to the problem. As I said before, **it's related to the instalation process.**

**Comment by mfriedrich74** [comment]
Those registry entries are not wrong. Relevant are those entries where
git_gui is referenced.
E.g. the mapping to extensions and file types.

**Comment by shadyueh** [comment]
Well, I've searched for any other entry about git_gui in the whole winreg and not found anything beyond those.

**Comment by dscho** [comment]
> I'm specifically talking about
> HKEY_CLASSES_ROOT\Directory\shell\git_gui\command whose value is
> "C:\Program Files\Git\cmd\git-gui.exe" "--working-dir" "%1"
> and
> HKEY_CLASSES_ROOT\Directory\shell\git_gui\command whose value is
> "C:\Program Files\Git\git-bash.exe" "--cd=%1"

That second one looks more like it should be `git_shell` instead of `git_gui`.

> By a reason that just dark deities know, this causes explorer replacement as default folders opener and we can't double click to open the folders.

That is *not* the effect here. I can double-click folders and the Explorer then opens the folder, as appropriate.

So I fear that this bug does not reproduce here, and most likely there is something else going on in your setup that my setup is missing.

**Comment by shadyueh** [comment]
I decided to use Portable version of git for windows while I was trying to figure out what was happening. After some research and tries due another error I was getting, I decided to clean my git installation and get the new brand version, 2.16.1.windows.4 (mine was 2.16.1.windows.2), and now everything seems working nicely. Researching in some forums I tried fix some reg entries too due the explorer replacement and found [this page](http://www.winhelponline.com/blog/file-asso-fixes-for-windows-7/). Maybe this helped me to fix any corrupted reg entry in win7 as I'm able to double-click folders and all. Thanks for your care and support. You're right, the problem doesn't seem related to those entries in reg, but maybe in my setup. I hope this help next people with similar problem.

**Comment by shadyueh** [comment]
I decided to use Portable version of git for windows while I was trying to figure out what was happening. After some research and tries due another error I was getting, I decided to clean my git installation and get the new brand version, 2.16.1.windows.4 (mine was 2.16.1.windows.2), and now everything seems working nicely. Researching in some forums I tried fix some reg entries too due the explorer replacement and found [this page](http://www.winhelponline.com/blog/file-asso-fixes-for-windows-7/). Maybe this helped me to fix any corrupted reg entry in win7 as I'm able to double-click folders and all. Thanks for your care and support. You're right, the problem doesn't seem related to those entries in reg, but maybe in my setup. I hope this help next people with similar problem.
